### DISCUSSION TITLE
Getting sudo-rs to build statically and work

### DISCUSSION THREAD

**Discussion by BlackDex** [BlackDex/rust-musl#7] [Q&A]
From this issue reported here: #4

I tried several ways to get sudo-rs to work, but had no luck.
If someone has an idea on how to do this, please let me know or create a PR.

I can build it just fine, but it always reports that a module is not found.

I tried linking multiple libraries in `build.rs` even tried to merge all `.a` files generated during the pam build into one large .a file.
Nothing seemed to be able to get it to work.

**Comment by BlackDex** [comment]
@Azathothas I have spoken to a developer of sudo-rs this weekend on FOSDEM.
He mentioned that PAM is a dynamically based system, and probably doesn't support statically all compiled in modules.

That is the reason why it doesn't work. But does work using your special script, which basically just zips all those modules and unpacks it and thus makes them still dynamically.

Therefor there is no way to have it working without separate modules.

**Reply by Azathothas** [reply]
Thanks for reaching out again.

Since the original issue, a lot has happened.
`staticx`, the tool I used to use package sudo-rs seems to have been abandoned and no longer maintained.
As a result, I haven't added `sudo-rs` to the new repo:https://github.com/pkgforge/soarpkgs
It only exists at the old, now archived repo: https://github.com/Azathothas/Toolpacks/blob/7fe47ab34b2648a2873d112f4e6b7e6423721a13/.github/scripts/x86_64_Linux/bins/sudo-rs.sh#L35

I have mentioned this issue at: https://github.com/leleliu008/ppkg-formula-repository-official-core/issues/14
The author of ppkg is much more experienced & hopefully we find a way to package sudo-rs in a portable way
